content/en/docs/concepts/storage/volumes.md

```diff
@@ -1000,469 +727,627 @@ backed by tmpfs (a RAM-backed filesystem) so they are never written to
-      fsType: ext4
-```
+You can set up your
+[PersistentVolume/PersistentVolumeClaim with raw block volume support](/docs/concepts/storage/persistent-volumes/#raw-block-volume-support) as usual, without any CSI specific changes.
 
-For more information, see the [vSphere volume](https://github.com/kubernetes/examples/tree/master/staging/volumes/vsphere) examples.
+#### CSI ephemeral volumes
 
-#### vSphere CSI migration {#vsphere-csi-migration}
+{{< feature-state for_k8s_version="v1.16" state="beta" >}}
 
-{{< feature-state for_k8s_version="v1.19" state="beta" >}}
+You can directly configure CSI volumes within the pod
+specification. Volumes specified in this way are ephemeral and do not
+persist across pod restarts. See [Ephemeral
+Volumes](/docs/concepts/storage/ephemeral-volumes/#csi-ephemeral-volume)
+for more information.
 
-The `CSIMigration` feature for `vsphereVolume`, when enabled, redirects all plugin operations
-from the existing in-tree plugin to the `csi.vsphere.vmware.com` {{< glossary_tooltip text="CSI" term_id="csi" >}} driver. In order to use this feature, the
-[vSphere CSI driver](https://github.com/kubernetes-sigs/vsphere-csi-driver)
-must be installed on the cluster and the `CSIMigration` and `CSIMigrationvSphere`
-[feature gates](/docs/reference/command-line-tools-reference/feature-gates/) must be enabled.
+For more information on how to develop a CSI driver, refer to the
+[kubernetes-csi documentation](https://kubernetes-csi.github.io/docs/).
 
-This also requires minimum vSphere vCenter/ESXi Version to be 7.0u1 and minimum HW Version to be VM version 15.
+#### Migrating to CSI drivers from in-tree plugins
 
-{{< note >}}
-The following StorageClass parameters from the built-in `vsphereVolume` plugin are not supported by the vSphere CSI driver:
+{{< feature-state for_k8s_version="v1.17" state="beta" >}}
 
-* `diskformat`
-* `hostfailurestotolerate`
-* `forceprovisioning`
-* `cachereservation`
-* `diskstripes`
-* `objectspacereservation`
-* `iopslimit`
+The `CSIMigration` feature, when enabled, directs operations against existing in-tree plugins to corresponding CSI plugins (which are expected to be installed and configured).  
+As a result, operators do not have to make any
+configuration changes to existing Storage Classes, PersistentVolumes or PersistentVolumeClaims
+(referring to in-tree plugins) when transitioning to a CSI driver that supersedes an in-tree plugin.
 
-Existing volumes created using these parameters will be migrated to the vSphere CSI driver,
-but new volumes created by the vSphere CSI driver will not be honoring these parameters.
-{{< /note >}}
+The operations and features that are supported include:
+provisioning/delete, attach/detach, mount/unmount and resizing of volumes.
 
-#### vSphere CSI migration complete {#vsphere-csi-migration-complete}
+In-tree plugins that support `CSIMigration` and have a corresponding CSI driver implemented are listed in [Types of Volumes](#volume-types).
```

Comment:
If an in-tree plugin doesn't support CSI migration it's listed separately? Why this distinction?

Reply:
I'll rephrase this, it's really just moving things that are superseded by a CSI approach to a separate section.